### DISCUSSION TITLE
[CasaOS] Permission denied to download torrent on external drive

### DISCUSSION THREAD

**Discussion by sharle4** [transmission/transmission#5849] [Q&A] [answered]
### What is the issue?

Hi,
I'm using **CasaOS** on a **Raspberry Pi**. I downloaded Transmission as a bittorent client.
All the os and the apps are installed on the **micro sd card of the raspberry pi**.
I plugged an **USB external hard drive** on my raspberry where I want the torrent to be downloaded.

Here is the setting i used for my docker container of Transmission :
![chrome_RMVMygTEod](https://github.com/transmission/transmission/assets/41182776/230853d6-c60a-46dc-b07b-6feecd0d7a07)
(where '_/media/devmon/Stockage Pi_' is the path of my mounted external hard drive")

The folders "complete" and "imcomplete" have been well created inside my dowload path indicated in the transmission settings 
![chrome_mP0oiWxGGc](https://github.com/transmission/transmission/assets/41182776/5994ca46-f351-4a77-b6ee-e2e2439f65a8)

The external hard drive seems to be well recognized in the Transmission settings (cf to the Free space)
![chrome_ZzDPKLQYA8](https://github.com/transmission/transmission/assets/41182776/b9197ebb-ced1-4928-877f-6a28fd95dff2)

However when I add a torrent i get this error message :
![Capture d'écran 2023-07-31 005126](https://github.com/transmission/transmission/assets/41182776/a6854677-deb8-40a0-82da-cf633ee34128)

I think then that my container dont have permission to write in the external drive, but i dont know why, how i could test that, how change that.

Thank you in advance for your help,
Charles

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

Transmission 3.00 (bb6b5a062e) installed on CasaOS (docker)

EDIT 1 :
This could be interesting : I cant access devmon folder without being super user in the terminal
![mintty_pLzu0JetTm](https://github.com/transmission/transmission/assets/41182776/7b805ed6-014e-46b0-b9e2-d3f436c91e92)

**Comment by tearfur** [comment]
Converting this to a discussion since this is not directly an issue with Transmission.

As for your question... Am I imagining things or is there no `/downloads` directory when you do `ls -al` in `/`?

**Comment by sharle4** [comment]
There is indeed no `/downloads` directory when I do `ls -al` in `/`. 
The reason is that my external hard drive is mounted in `/media/devmon/Stockage Pi` (and as i shown it in my last screen, the folder `devmon` seems to bee unaccessible for a commun user)
The `/downloads` folder is (at least as i understand it) a label for the path `/media/devmon/Stockage Pi` as I set it up in the settings of the Transmission app (and it looks understood by Transmission since folders _complete_ and _uncomplete_ have been created in the path `/media/devmon/Stockage Pi`

**Reply by tearfur** [reply]
Ah right, `/downloads` is a docker volume... In that case, I think I remember from somewhere (don't quote me), that the Docker container directly uses the file permissions of the host machine for Docker volumes. For example, if you are trying to access a Docker volume with the user `sharle4` within the container, then you must also grant a user named `sharle4` the necessary file permissions *on the host machine*.

**Reply by sharle4** [reply]
Thank you for digging deep into your memory. However, the issue I encounter is that there is only one user on casaos, so i cant figure out how i'm supposed to grant necessary permissions to transmission.

**Reply by tearfur** [reply]
You just create a user with the same name in the host machine and grant permissions to that user on the host machine.

**Comment by sharle4** [comment, marked as answer]
I finally solved this issue by
1 : formatting my external hard drive in ext4
2 : running the command sudo `chown -R pi:pi /media/devmon/StockagePi`
thats it
Thank you very much for your help and your time @tearfur <3

**Reply by cascafico** [reply]
Since my external disk is filled with 1 TB of data, I preferred to fix mounting options:
1) find unique disk id (UUID) 
`    ls -l /dev/disk/by-uuid/
`    
2) add mounting options in /etc/fstab (gid and uid 1000 for pi)
`   UUID=<UUID found in step 1> /media/devmon/Ext_1G ntfs-3g umask=0022,gid=1000,uid=1000
`   
3) if needed:
`   chown pi:pi /media/devmon
`

**Comment by oniricProject** [comment]
Hi 
Noob here, but I have a related issue and I'm wondering if there is an easier solution.
I installed CasaOs on a Raspberry Pi 4B, then Plex and then tried to make it work with a 1TB USB drive. NTFS. It worked and I could modify the USB folders no problem, even upload stuff from my computer. That was just a try. I then decided to switch the 1TB to a 4TB hard drive (NTFS) in the same way, which was residing inside my computer. Unfortunately this drive seems to work read-only, on CasaOs, so I cannot seem to upload new contents to it.
Inspecting the two hard drives the only difference I managed to find is permissions. on the 1TB drive "everyone" was missing. I tried to set the same permissions on the 4TB one but it didn't change anything. 
I'm wondering if there is any way I can set the r/w propertied on the 4TB drive without having to format it to the ext4 file system.

**Comment by mihhail80** [comment]
how to enable uploading torrents on qbittorrent?
